### DISCUSSION TITLE
planetscale says drizzle is supported, but first test query produces "VT12001: unsupported: lateral derived tables" error

### DISCUSSION THREAD

**Discussion by ortonomy** [planetscale/discussion#573] [Bug Reports] [answered]
I'm using drizzle with relatively uncomplicated self reference table + fk + relation

table def

```
export const User = mysqlTable('users', {
  userId: bigint('user_id', { mode: 'number' }).primaryKey().autoincrement(),
  email: varchar('email_address', { length: 255 }).unique().notNull(),
  username: varchar('username', { length: 255 }).unique(),
  ...Timestamps,
  createdBy: bigint('created_by', { mode: 'number' }).references((): AnyMySqlColumn => User.userId),
  updatedBy: bigint('updated_by', { mode: 'number' }).references((): AnyMySqlColumn => User.userId),
  deletedBy: bigint('deleted_by', { mode: 'number' }).references((): AnyMySqlColumn => User.userId)
})
```

relation def

```
export const UserAuditRelations = relations(User, ({ one }) => ({
  creator: one(User, {
    fields: [User.createdBy],
    references: [User.userId]
  }),
  updater: one(User, {
    fields: [User.updatedBy],
    references: [User.userId]
  }),
  remover: one(User, {
    fields: [User.deletedBy],
    references: [User.userId]
  })
}))
```

simple list query

```
ctx.dbClient.query.User.findMany({ with: { creator: true } })
```

results in sql like this from Drizzle (*not sure why it's creating a later join or usung json array, but there you go)

```
Query: select `User`.`user_id`, `User`.`email_address`, `User`.`username`, `User`.`created_at`, `User`.`updated_at`, `User`.`deleted_at`, `User`.`created_by`, `User`.`updated_by`, `User`.`deleted_by`, `User_creator`.`data` as `creator` from `users` `User` left join lateral (select json_array(`User_creator`.`user_id`, `User_creator`.`email_address`, `User_creator`.`username`, `User_creator`.`created_at`, `User_creator`.`updated_at`, `User_creator`.`deleted_at`, `User_creator`.`created_by`, `User_creator`.`updated_by`, `User_creator`.`deleted_by`) as `data` from (select * from `users` `User_creator` where `User_creator`.`user_id` = `User`.`created_by` limit ?) `User_creator`) `User_creator` on true -- params: [1]
```

But planetscale returns error:

`VT12001: unsupported: lateral derived tables`

which suggests a) planetscale doesn't actually properly support mysql  despite saying 'is the world’s most advanced MySQL platform' b) isn't actually compatible with drizzle, despite being listed as a client option in setup. 

Thoughts?

**Comment by janpio** [comment, marked as answer]
@prisma here, but as I know a bit of Drizzle as well, I can help:

https://orm.drizzle.team/docs/rqb/#modes

With this the Drizzle relation queries should also work with PlanetScale.

**Reply by ortonomy** [reply]
Ah, thanks. @janpio  - I will test this. I did *not* expect to have to configure this to work with different services. Can we change the front-page of planetscale to by say "advanced MySQL(not vanilla) platform'

(Glad you felt the need to plug Prisma, though. I don't want to use prisma because it makes me write a DSL for its schema and that is A Very Bad Idea TM)

**Reply by janpio** [reply]
I did not feel the need to _plug_ Prisma, but the need to _disclose_ that I work for Prisma and am _still_ being helpful when questions about Drizzle are being asked. You might want to reconsider your reaction (but of course can prefer Drizzle, they do things differently for a reason and that makes them a better match for some).
